src/lib/network_pool/snark_pool.ml

```diff
@@ -252,53 +255,91 @@ struct
           | Some _ ->
               true )
 
-      let fee_is_sufficient t ~fee ~prover ~best_tip_ledger =
-        let open Mina_base in
-        Currency.Fee.(fee >= t.account_creation_fee)
-        ||
-        match best_tip_ledger with
-        | None ->
-            false
-        | Some l ->
-            Option.(
-              is_some
-                ( Base_ledger.location_of_account l
-                    (Account_id.create prover Token_id.default)
-                >>= Base_ledger.get l ))
+      let fee_is_sufficient t ~fee ~account_exists =
+        Currency.Fee.(fee >= t.account_creation_fee) || account_exists
 
-      let handle_transition_frontier_diff u t =
-        match u with
-        | `New_best_tip l ->
-            Statement_table.filteri_inplace t.snark_tables.all
-              ~f:(fun ~key ~data:{fee= {fee; prover}; _} ->
-                let keep =
-                  fee_is_sufficient t ~fee ~prover ~best_tip_ledger:(Some l)
+      let handle_new_best_tip_ledger t ledger =
+        let open Mina_base in
+        let open Signature_lib in
+        trace_recurring "handle_new_best_tip_ledger" (fun () ->
+            Throttle.enqueue t.snark_table_lock (fun () ->
+                let%bind prover_account_ids =
```

Comment:
Now that this is async, shouldn't this be in `Interruptible.t` and hang off the ledger's detach interrupt? Otherwise, we might find that the ledger has gone away, especially during catchup.